tests/test_py_structures_inputs.py

```diff
@@ -0,0 +1,526 @@
+    GeometryList,
+    GriddedSamples,
+    InputError,
+    MakeGridParameters,
+    Mesh1d,
+    Mesh2d,
+    MeshKernel,
+    MeshRefinementParameters,
+    OrthogonalizationParameters,
+    RefinementType,
+    SplinesToCurvilinearParameters,
+)
+
+
+def test_mesh2d_implicit_int_conversions():
+    """Test implicit conversion from int to double for Mesh2d works"""
+    mk = MeshKernel()
+
+    edge_nodes = np.array([0, 1, 1, 2, 2, 3, 3, 0], dtype=np.int32)
+    node_x = np.array([0, 1, 1, 0], dtype=np.int32)
+    node_y = np.array([0, 0, 1, 1], dtype=np.int32)
+
+    input_mesh2d = Mesh2d(node_x, node_y, edge_nodes)
+    mk.mesh2d_set(input_mesh2d)
+
+    output_mesh2d = mk.mesh2d_get()
+
+    # Test if the input and output differs
+    assert_array_equal(output_mesh2d.edge_nodes, input_mesh2d.edge_nodes)
+    assert_array_equal(output_mesh2d.node_x, input_mesh2d.node_x)
+    assert_array_equal(output_mesh2d.node_y, input_mesh2d.node_y)
+
+    # Test if faces are correctly calculated
+    assert_array_equal(output_mesh2d.face_nodes, np.array([0, 1, 2, 3]))
+    assert_array_equal(output_mesh2d.nodes_per_face, np.array([4]))
+    assert_array_equal(output_mesh2d.face_x, np.array([0.5]))
+    assert_array_equal(output_mesh2d.face_y, np.array([0.5]))
+
+
+def test_mesh2d_implicit_string_conversions():
+    """Test implicit conversions from string to doubles works for Mesh2d"""
+    node_x = ["1.0", "2.0", "3.0"]
+    node_y = ["4.0", "5.0", "6.0"]
+    edge_nodes = ["0", "1", "1", "2"]
+    face_nodes = ["0", "1", "2"]
+    nodes_per_face = ["3", "4", "5"]
+    edge_x = ["1.5", "2.5"]
+    edge_y = ["4.5", "5.5"]
+    face_x = ["2.0", "3.0"]
+    face_y = ["5.0", "6.0"]
+    edge_faces = ["0", "1"]
+    face_edges = ["2", "3"]
+
+    mesh_2d = Mesh2d(
+        node_x=node_x,
+        node_y=node_y,
+        edge_nodes=edge_nodes,
+        face_nodes=face_nodes,
+        nodes_per_face=nodes_per_face,
+        edge_x=edge_x,
+        edge_y=edge_y,
+        face_x=face_x,
+        face_y=face_y,
+        edge_faces=edge_faces,
+        face_edges=face_edges,
+    )
+
+    assert_array_equal(mesh_2d.node_x, np.array([1.0, 2.0, 3.0], dtype=np.double))
+    assert_array_equal(mesh_2d.node_y, np.array([4.0, 5.0, 6.0], dtype=np.double))
+    assert_array_equal(mesh_2d.edge_nodes, np.array([0, 1, 1, 2], dtype=np.int32))
+    assert_array_equal(mesh_2d.face_nodes, np.array([0, 1, 2], dtype=np.int32))
+    assert_array_equal(mesh_2d.nodes_per_face, np.array([3, 4, 5], dtype=np.int32))
+    assert_array_equal(mesh_2d.edge_x, np.array([1.5, 2.5], dtype=np.double))
+    assert_array_equal(mesh_2d.edge_y, np.array([4.5, 5.5], dtype=np.double))
+    assert_array_equal(mesh_2d.face_x, np.array([2.0, 3.0], dtype=np.double))
+    assert_array_equal(mesh_2d.face_y, np.array([5.0, 6.0], dtype=np.double))
+    assert_array_equal(mesh_2d.edge_faces, np.array([0, 1], dtype=np.int32))
+    assert_array_equal(mesh_2d.face_edges, np.array([2, 3], dtype=np.int32))
```

Comment:
Optional: The expected array can be `np.asarray(input, dtype=np.double)` rather explicitly creating an array with the expected dtype. `input` is one of the vars defined in lines 50-60.